azure-quantum/azure/quantum/target/target.py

```diff
@@ -149,7 +150,10 @@ def submit(
         :return: Azure Quantum job
         :rtype: Job
         """
-        input_params = input_params or {}
+        if isinstance(input_params, InputParams):
+            input_params = input_params.as_dict()
+        else:
```

Comment:
There's some changes here from XField. You might need to resolve some merge conflicts.

Reply:
I paid attention during merging.